pkg/api/core/handler_api.go

```diff
@@ -253,6 +257,11 @@ func AttachRoutes(handlers *Handlers) http.Handler {
 			r.Method(http.MethodPost, "/", handlers.CreateCluster)
 			r.Method(http.MethodDelete, "/", handlers.DeleteCluster)
 		})
+
+		r.Route("/clustersecuritygroupid", func(r chi.Router) {
```

Comment:
That was my inital thought. But I was unsure how RESTful that would be, as I haven't seen such an endpoint before. The normal REST way would be:
/clusters/{clusterId}/securitygroupid.

... which we can't do, since don't have a clusterId, only the ID struct, which consists of three fields.

Since we pass a struct instead of just a clusterId string, perhaps your suggestion is okay.

Reply:
Yeah. It makes sense to me atleast to stick to the RESTful style, but with parameters in the structs instead. At least that will give us a consistent style to follow

Reply:
/clustersecuritygroupid doesn't scale so well, i.e. if all sorts of attributes are put on root level, it's going to be messy. So I'll go for the non-RESTful /clusters/securitygroupid.